From 9cc367a2002b6c550b467e0471160fb610968e3d Mon Sep 17 00:00:00 2001
From: eric <eric.peterson1@twcable.com>
Date: Mon, 8 Dec 2014 16:38:26 -0700
Subject: [PATCH] Horizon login page contains DOS attack mechanism

the horizon login page (and middleware) accesses the session
too early in the login process, which will create session records
in the session backend.  This is especially problematic when non-cookie
backends are used.

Co-Authored-By: Tihomir Trifonov <t.trifonov@gmail.com>
Co-Authored-By: Eric Peterson <eric.peterson1@twcable.com>

Change-Id: I9a4999eb5f053515575ef09b8ba9d3bb3f114e5c
Closes-Bug: 1394370
(cherry picked from commit e676c88a329af57d6c4f13df54f6e1e06c1f8360)
---
 openstack_auth/forms.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/openstack_auth/forms.py b/openstack_auth/forms.py
index 2c8092c..8c1fcee 100644
--- a/openstack_auth/forms.py
+++ b/openstack_auth/forms.py
@@ -98,7 +98,6 @@ class Login(django_auth_forms.AuthenticationForm):
             msg = 'Login failed for user "%(username)s".' % \
                 {'username': username}
             LOG.warning(msg)
-            self.request.session.flush()
             raise forms.ValidationError(exc)
         if hasattr(self, 'check_for_test_cookie'):  # Dropped in django 1.7
             self.check_for_test_cookie()
